Assets/LeapMotionModules/UI/Interaction/InteractionBehaviour.cs

```diff
@@ -0,0 +1,692 @@
+    /// target position, ignoring collisions. Inherit will simply use the isKinematic
+    /// state of the Rigidbody from before it was grasped.
+    /// </summary>
+    public enum GraspedMovementType {
+      Inherit,
+      Kinematic,
+      Nonkinematic
+    }
+    [DisableIf("moveObjectWhenGrasped", isEqualTo: false)]
+    [Tooltip("When the object is moved by a holding InteractionHand, how should it move to its "
+           + "new position? Nonkinematic bodies will collide with other Rigidbodies, so they "
+           + "might not reach the target position. Kinematic rigidbodies will always move to the "
+           + "target position, ignoring collisions. Inherit will simply use the isKinematic "
+           + "state of the Rigidbody from before it was grasped.")]
+    public GraspedMovementType graspedMovementType;
+
+    /// <summary> The RigidbodyWarper manipulates the graphical (but not physical) position
+    /// of grasped objects based on the movement of the Leap hand so they appear move with less latency. </summary>
+    /// TODO: This is not actually implemented.
+    [HideInInspector]
+    public RigidbodyWarper rigidbodyWarper;
+
+    [Header("Advanced Settings")]
+    [Tooltip("Warping manipulates the graphical (but not physical) position of grasped objects "
+           + "based on the movement of the Leap hand so the objects appear to move with less latency.")]
+    public bool graspHoldWarpingEnabled__curIgnored = true; // TODO: Warping not yet implemented.
+
+    protected override void Awake() {
+      base.Awake();
+      rigidbodyWarper = new RigidbodyWarper(interactionManager, this.transform, _body, 0.25F);
+    }
+
+    protected override void Start() {
+      base.Start();
+
+      RefreshPositionLockedState();
+      InitGrasping();
+
+      InitInternal();
+    }
+
+    /// <summary>
+    /// InteractionManager manually calls this directly
+    /// after all InteractionHands are updated (in FixedUpdate).
+    /// 
+    /// Hovering uses its update to provide per-object (as opposed to per-hand)
+    /// hover callbacks, e.g. OnObjectHoverStay(), which fires once per frame
+    /// while the object is hovered by any number of hands greater than zero.
+    /// 
+    /// Grasping uses its update to provide per-object (as opposed to per-hand)
+    /// grasp callbacks, e.g. OnObjectGraspBegin(), similarly. It will also
+    /// fire multi-grasp callbacks, e.g. OnMultiGraspBegin(), but only if
+    /// multiHandGrasping is enabled.
+    /// </summary>
+    public override void FixedUpdateObject() {
+      FixedUpdateHovering();
+      FixedUpdatePrimaryHovering();
+      FixedUpdateContact();
+      FixedUpdateGrasping();
+
+      FixedUpdateCollisionMode();
+    }
+
+    // TODO: Currently this gets the distance from the point to this transform, but this will
+    // need to incorporate distance to the rigidbody.
+    private float GetInteractionDistanceToPoint(Vector3 point) {
```

Comment:
Would be nice if this was protected virtual to allow you to specify a custom metric as well.